### ISSUE TITLE
complex-atanh signals error

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2720]
Imported from SourceForge on 2024-07-06 18:16:15
Created by **[rtoy](https://sourceforge.net/u/rtoy/)** on 2024-05-06 15:54:30
Original: https://sourceforge.net/p/maxima/bugs/4299

---

`complex-atanh` signals a weird error:
```
(%i1) :lisp (complex-atanh '((BIGFLOAT SIMP 56) -48685849814366015 -1) '((BIGFLOAT SIMP 56) 0 0))

COMPLEX-ATANH: HOW DID I GET HERE?
 -- an error. To debug this try: debugmode(true);
```
This is happening in `complex-atanh` in the computation of the value of `nu`.  I'm not really sure what's happening but the comment hints at the problem:
```
;; When y = 0, Im atanh z = 1/2 (arg(1 + x) - arg(1 - x))
;;                        = if x < -1 then %pi/2 else if x > 1 then -%pi/2 else <whatever>
```

In this case, `y = 0`, but `|x| < 1`, so the `<whatever>` case holds. This is where merror is called.

If the formula is correct, when `|x| < 1`, `arg(1+x) - arg(1-x)` is 0. so `<whatever>` should be 0.

(BTW, the code is really poorly indented and things don't line up. Yuck!)

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 18:16:17
Created by **[rtoy](https://sourceforge.net/u/rtoy/)** on 2024-05-08 21:18:41
Original: https://sourceforge.net/p/maxima/bugs/4299/#7a4a

---

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 18:16:20
Created by **[rtoy](https://sourceforge.net/u/rtoy/)** on 2024-05-08 21:18:41
Original: https://sourceforge.net/p/maxima/bugs/4299/#5240

---

Fixed in commit [8a19e9ecd2ce8ec7c1c04ba7129e46103529d086]
